src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,199 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "ValidLogin";
+    private static final String MIN_LENGTH_VALID = "6chars";
+    private static final String MIN_EDGE_LENGTH_VALID = "7 chars";
+    private static final String MAX_EDGE_LENGTH_INVALID = "5char";
+    private static final String IS_EMPTY = "";
+    private static final String IS_BLANK = "      ";
+    private static final String VALID_PASSWORD = "ValidPassword";
+    private static final int VALID_AGE = 19;
+    private static final int MIN_VALID_AGE = 18;
+    private static final int MAX_INVALID_AGE = 17;
+    private static final int AGE_IS_ZERO = 0;
+    private static final int AGE_IS_NEGATIVE = -18;
+    private static RegistrationService registrationService;
+    private static User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        user = new User();
+    }

```

Comment:
I don't think there is a necessity of using `@BeforeEach` with static variables. In addition - you don't need to declare some generic `User`, since you should prepare those users per each test separately - you're testing `RegistrationService` here

why not just:
```suggestion
    private final RegistrationService registrationService = new RegistrationServiceImpl();
```